protos/action_server/action_server.proto

```diff
@@ -0,0 +1,185 @@
+}
+
+message SetAllowableFlightModeRequest
+{
+    AllowableFlightModes flight_modes = 1;
+}
+
+message GetAllowableFlightModeRequest {}
+
+message SubscribeArmDisarmRequest {}
+
+message SubscribeFlightModeChangeRequest {}
+
+message SubscribeTakeoffRequest {}
+
+message SubscribeLandRequest {}
+
+message SubscribeRebootRequest {}
+
+message SubscribeShutdownRequest {}
+
+message SubscribeTerminateRequest {}
+
+message ArmDisarmResponse {
+    ActionServerResult action_server_result = 1;
+    ArmDisarm arm = 2;
+}
+
+message FlightModeChangeResponse {
+    ActionServerResult action_server_result = 1;
+    FlightMode flight_mode = 2;
+}
+
+message TakeoffResponse {
+    ActionServerResult action_server_result = 1;
+    bool takeoff = 2;
+}
+
+message LandResponse {
+    ActionServerResult action_server_result = 1;
+    bool land = 2;
+}
+
+message RebootResponse {
+    ActionServerResult action_server_result = 1;
+    bool reboot = 2;
+}
+
+message ShutdownResponse {
+    ActionServerResult action_server_result = 1;
+    bool shutdown = 2;
+}
+
+message TerminateResponse {
+    ActionServerResult action_server_result = 1;
+    bool terminate = 2;
+}
+
+message SetArmableResponse {
+    ActionServerResult action_server_result = 1;
+}
+
+message SetDisarmableResponse {
+    ActionServerResult action_server_result = 1;
+}
+
+message SetAllowableFlightModeResponse {
+    ActionServerResult action_server_result = 1;
+}
+
+message SetAllowTakeoffResponse {
+    ActionServerResult action_server_result = 1;
+}
+
+message GetAllowableFlightModeResponse {
+    AllowableFlightModes flight_modes = 1;
+}
+
+message AllowableFlightModes {
+    bool can_auto_mode = 1; // Auto/mission mode 
+    bool can_guided_mode = 2; // Guided mode
+    bool can_stabilize_mode = 3; // Stabilize mode
+}
+
+// Arming message type
+message ArmDisarm {
+    int32 arm = 1; // Should vehicle arm
+    int32 force = 2; // Should arm override pre-flight checks
+}
+
+/*
+ * Flight modes.
+ *
+ * For more information about flight modes, check out
+ * https://docs.px4.io/master/en/config/flight_mode.html.
+ */
+enum FlightMode {
+    FLIGHT_MODE_UNKNOWN = 0; // Mode not known
+    FLIGHT_MODE_READY = 1; // Armed and ready to take off
+    FLIGHT_MODE_TAKEOFF = 2; // Taking off
+    FLIGHT_MODE_HOLD = 3; // Holding (hovering in place (or circling for fixed-wing vehicles)
+    FLIGHT_MODE_MISSION = 4; // In mission
+    FLIGHT_MODE_RETURN_TO_LAUNCH = 5; // Returning to launch position (then landing)
+    FLIGHT_MODE_LAND = 6; // Landing
+    FLIGHT_MODE_OFFBOARD = 7; // In 'offboard' mode
+    FLIGHT_MODE_FOLLOW_ME = 8; // In 'follow-me' mode
+    FLIGHT_MODE_MANUAL = 9; // In 'Manual' mode
+    FLIGHT_MODE_ALTCTL = 10; // In 'Altitude Control' mode
+    FLIGHT_MODE_POSCTL = 11; // In 'Position Control' mode
+    FLIGHT_MODE_ACRO = 12; // In 'Acro' mode
+    FLIGHT_MODE_STABILIZED = 13; // In 'Stabilize' mode
+    FLIGHT_MODE_RATTITUDE = 14; // In 'Rattitude' mode
```

Comment:
Actually RATTITUDE has been removed, so that's the first one that you can save :smile:.

Reply:
RATs...